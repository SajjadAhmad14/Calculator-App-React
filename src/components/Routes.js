import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link,
} from 'react-router-dom';
import App from './App';
// import Home from './Home';
import Qoutes from './Qoutes';

const Routes = () => (
  <Router>
    <div className="nav">
      <Link to="/">
        <h2>Maths-Up</h2>
      </Link>
      <ul>
        <Link to="/qoutes">
          <li>Qoutes</li>
        </Link>
        <Link to="/calculator">
          <li>Calculator</li>
        </Link>
      </ul>
    </div>
    <Switch>
      <Route exact path="/qoutes" component={Qoutes} />
      <Route exact path="/calculator" component={App} />
    </Switch>
  </Router>
);

export default Routes;
