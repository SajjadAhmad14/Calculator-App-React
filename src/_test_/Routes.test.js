import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';
import Routes from '../components/Routes';

describe('Test Routes component', () => {
  beforeEach(() => {
    render(<Routes />)
  });

  it ('Add screenshot test', () => {
    const { asFragment } = render(<Routes />)
    expect(asFragment(<Routes />)).toMatchSnapshot();
  })

  it('Display calculator', () => {
    const text = screen.getByText(/Calculator/i);
    expect(text).toBeInTheDocument();
  })

  it('Display qoutes', () => {
    const text = screen.getByText(/Qoutes/i);
    expect(text).toBeInTheDocument();
  })

  it('Display Maths up', () => {
    const text = screen.getByText(/Maths-up/i);
    expect(text).toBeInTheDocument();
  })

})