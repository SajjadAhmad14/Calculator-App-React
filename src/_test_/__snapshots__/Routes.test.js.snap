// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Routes component Add screenshot test 1`] = `
<DocumentFragment>
  <div
    class="nav"
  >
    <a
      href="/"
    >
      <h2
        data-testid="maths"
      >
        Maths-Up
      </h2>
    </a>
    <ul>
      <a
        href="/qoutes"
      >
        <li>
          Qoutes
        </li>
      </a>
      <a
        href="/calculator"
      >
        <li>
          Calculator
        </li>
      </a>
    </ul>
  </div>
  <div
    class="app"
  >
    <div
      class="display-board"
    >
      <h1
        class="result"
      >
        null
      </h1>
    </div>
    <div
      class="btn-panel"
    >
      <div
        class="group"
      >
        <button
          class="btn-width"
          type="button"
        >
          AC
        </button>
        <button
          class="btn-width"
          type="button"
        >
          +/-
        </button>
        <button
          class="btn-width"
          type="button"
        >
          %
        </button>
        <button
          class="btn-width orange"
          type="button"
        >
          /
        </button>
      </div>
      <div
        class="group"
      >
        <button
          class="btn-width"
          type="button"
        >
          7
        </button>
        <button
          class="btn-width"
          type="button"
        >
          8
        </button>
        <button
          class="btn-width"
          type="button"
        >
          9
        </button>
        <button
          class="btn-width orange"
          type="button"
        >
          X
        </button>
      </div>
      <div
        class="group"
      >
        <button
          class="btn-width"
          type="button"
        >
          4
        </button>
        <button
          class="btn-width"
          type="button"
        >
          5
        </button>
        <button
          class="btn-width"
          type="button"
        >
          6
        </button>
        <button
          class="btn-width orange"
          type="button"
        >
          -
        </button>
      </div>
      <div
        class="group"
      >
        <button
          class="btn-width"
          type="button"
        >
          1
        </button>
        <button
          class="btn-width"
          type="button"
        >
          2
        </button>
        <button
          class="btn-width"
          type="button"
        >
          3
        </button>
        <button
          class="btn-width orange"
          type="button"
        >
          +
        </button>
      </div>
      <div
        class="group"
      >
        <button
          class="width-50"
          type="button"
        >
          0
        </button>
        <button
          class="btn-width"
          type="button"
        >
          .
        </button>
        <button
          class="btn-width orange"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
